src/commonMain/kotlin/org/treeWare/model/operator/Difference.kt

```diff
@@ -0,0 +1,297 @@
+        val newField = leaderFieldList.last()
+
+        val createParent = createStack.first() as MutableBaseEntityModel
+        val createField = newDisconnectedField(createParent, lastLeaderFieldName)
+        createStack.addFirst(createField)
+
+        val deleteParent = deleteStack.first() as MutableBaseEntityModel
+        val deleteField = newDisconnectedField(deleteParent, lastLeaderFieldName)
+        deleteStack.addFirst(deleteField)
+
+        val updateParent = updateStack.first() as MutableBaseEntityModel
+        val updateField = newDisconnectedField(updateParent, lastLeaderFieldName)
+        updateStack.addFirst(updateField)
+
+        /**===== Check if it's absent in the original =====**/
+        if (oldField == null) { //if the original doesn't have the entity:
+            elementInclusion.inCreate = true //this should be in the create tree
+            copy(checkNotNull(newField), createField)
+            return TraversalAction.ABORT_SUB_TREE
+        }
+
+        /**===== Check if it's absent in the new version =====**/
+        if (newField == null) {//if the new version doesn't have the entity:
+            elementInclusion.inDelete = true //this should be in the delete tree
+            copy(oldField, deleteField)
+            return TraversalAction.ABORT_SUB_TREE
+        }
+
+        return TraversalAction.CONTINUE
+    }
+
+    private fun leaveField(leaderFieldList: List<FieldModel?>) {
+        val elementInclusion = inclusionStack.removeFirst()
+
+        /**===== Handle the create tree =====**/
+        val createField = createStack.removeFirst() as MutableFieldModel
+        if ((elementInclusion.inCreate) || isKeyField(createField)) { //Case: In or above a created node
+            val createParent = createStack.first() as MutableBaseEntityModel
+            //(createField as MutableSingleFieldModel).setValue()
+            connectField(createParent, createField)
+        }
+
+        /**===== Handle the delete tree =====**/
+        val deleteField = deleteStack.removeFirst() as MutableFieldModel
+        if ((elementInclusion.inDelete) || isKeyField(deleteField)) { //Case: In or above a deleted node
+            val deleteParent = deleteStack.first() as MutableBaseEntityModel
+            connectField(deleteParent, deleteField)
+        }
+
+        /**===== Handle the update tree =====**/
+        val updateField = updateStack.removeFirst() as MutableFieldModel
+        if ((elementInclusion.inUpdate) || isKeyField(updateField)) { //Case: In or above an updated node
+            val updateParent = updateStack.first() as MutableBaseEntityModel
+            connectField(updateParent, updateField)
+        }
+
+        inclusionStack.first().getChildInclusions(elementInclusion)
+    }
+}
+
+private data class InclusionData(
+    var inCreate: Boolean = false,
+    var inDelete: Boolean = false,
+    var inUpdate: Boolean = false
+) {
+    fun getChildInclusions(child: InclusionData) {
```

Comment:
nit: since this method is not returning anything, the `get` prefix does not sound right. `add` prefix might be better (`addChildInclusions`)